src/OpenLoco/src/GameCommands/Road/RemoveRoad.cpp

```diff
@@ -0,0 +1,260 @@
+            if (elRoad->mods())
+            {
+                return nullptr;
+            }
+
+            return elRoad;
+        }
+
+        return nullptr;
+    }
+
+    // 0x00477A10
+    static currency32_t roadRemoveCost(const RoadRemovalArgs& args, const World::TrackData::PreviewTrack roadPiece0, const World::Pos3 roadStart, const uint8_t flags)
+    {
+        const auto roadLoc = roadStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ roadPiece0.x, roadPiece0.y }, args.rotation), roadPiece0.z };
+
+        auto tile = World::TileManager::get(roadLoc.x, roadLoc.y);
+        auto* roadElPiece = getRoadElement(tile, args, roadPiece0.index, flags);
+        if (roadElPiece == nullptr)
+        {
+            return 0;
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        const auto* roadObj = ObjectManager::get<RoadObject>(roadElPiece->roadObjectId());
+        {
+            const auto trackBaseCost = Economy::getInflationAdjustedCost(roadObj->sellCostFactor, roadObj->costIndex, 10);
+            const auto cost = (trackBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+            totalRemovalCost += cost;
+        }
+
+        // Check mod removal costs
+        for (auto i = 0U; i < 2; i++)
+        {
+            if (roadElPiece->hasMod(i))
+            {
+                const auto* trackExtraObj = ObjectManager::get<RoadExtraObject>(roadObj->mods[i]);
+                const auto trackExtraBaseCost = Economy::getInflationAdjustedCost(trackExtraObj->sellCostFactor, trackExtraObj->costIndex, 10);
+                const auto cost = (trackExtraBaseCost * World::TrackData::getTrackMiscData(args.roadId).costFactor) / 256;
+                totalRemovalCost += cost;
+            }
+        }
+
+        return totalRemovalCost;
+    }
+
+    // 0x004775A5
+    static uint32_t removeRoad(const RoadRemovalArgs& args, uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3{ 16, 16, 0 });
+
+        // 0x0047762D
+        auto tile = World::TileManager::get(args.pos.x, args.pos.y);
+
+        // This part was made redundant by a no-op write in Hooks.cpp
+        // writeNop(0x004776DD, 6);
+        /*
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
+            if (!sub_431E6A(companyId, reinterpret_cast<const World::TileElement*>(&elRoad)))
+            {
+                return FAILURE;
+            }
+
+            setErrorText(StringIds::empty);
+
+            auto nearest = TownManager::getClosestTownAndDensity(args.pos);
+            if (nearest.has_value())
+            {
+                auto* town = TownManager::get(nearest->first);
+                FormatArguments::common(town->name);
+                setErrorText(StringIds::stringid_local_authority_wont_allow_removal_in_use);
+                return FAILURE;
+            }
+        }
+        */
```

Comment:
I think we should probably turn this into a proper setting, rather than omitting it outright? Then again, it's been like this since 2018...